grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -592,22 +555,34 @@ public RuleType putRuleType(TypeLabel label) {
 
     @Override
     public <T extends Concept> T getConcept(ConceptId id) {
-        if(getConceptLog().isConceptCached(id)){
-            return getConceptLog().getCachedConcept(id);
+        if(getTxCache().isConceptCached(id)){
+            return getTxCache().getCachedConcept(id);
         } else {
             return getConcept(Schema.ConceptProperty.ID, id.getValue());
         }
     }
-    private <T extends Type> T getTypeByLabel(TypeLabel label, Schema.BaseType baseType){
-        Type type = buildType(label, ()->getConcept(Schema.ConceptProperty.TYPE_LABEL, label.getValue()));
+    private <T extends Type> T getType(TypeLabel label, Schema.BaseType baseType){
+        operateOnOpenGraph(() -> null); //Makes sure the graph is open
+
+        Type type = buildType(label, ()-> {
+            Integer id = convertToId(label);
+            if (id != -1) {
+                return getType(id);
+            } else {
+                return null;
+            }
```

Comment:
See my comment above about turning it into an Optional. Happy to do it if I dont end up passing in magic number into the graql traversal.